### PR TITLE
Updated View and Template for Population type in active on ART patient linelist

### PR DESCRIPTION
**Pull request by patryllus**
pop_type

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/PopulationTypeActiveDataDefinition.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import org.openmrs.module.reporting.data.BaseDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PersonDataDefinition;
+import org.openmrs.module.reporting.definition.configuration.ConfigurationPropertyCachingStrategy;
+import org.openmrs.module.reporting.evaluation.caching.Caching;
+
+/**
+ * Population Type Data Definition
+ */
+@Caching(strategy=ConfigurationPropertyCachingStrategy.class)
+public class PopulationTypeActiveDataDefinition extends BaseDataDefinition implements PersonDataDefinition {
```
Comment (makombe) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/PopulationTypeActiveDataDefinition.java:21]: I think you can name the class properly,does PopulationTypeActive makes sense...feels like it is hanging

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/PopulationTypeActiveDataEvaluator.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Map;
+
+/**
+ * Evaluates a VisitIdDataDefinition to produce a VisitData
+ */
+@Handler(supports=PopulationTypeActiveDataDefinition.class, order=50)
+public class PopulationTypeActiveDataEvaluator implements PersonDataEvaluator {
```
Comment (makombe) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/PopulationTypeActiveDataEvaluator.java:29]: Same here for the evaluator naming of the class

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/PopulationTypeActiveDataEvaluator.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Map;
+
+/**
+ * Evaluates a VisitIdDataDefinition to produce a VisitData
```
Comment (makombe) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/PopulationTypeActiveDataEvaluator.java:26]: Make the comment reflect what the evaluator is all about

**Comment by makombe** [comment]
@patryllus  Thanks for attaching the screenshot. That is awesome.

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/PopulationTypeActiveDataDefinition.java (hunk 1 above)
Comment (patryllus) [reply api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/PopulationTypeActiveDataDefinition.java:21]: Its an Active patients line list cohort. But let me call it ActivePatientsPopulationType    does that make sense?

File: api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/ActivePatientsPopulationTypeDataEvaluator.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+@Handler(supports=ActivePatientsPopulationTypeDataDefinition.class, order=50)
+public class ActivePatientsPopulationTypeDataEvaluator implements PersonDataEvaluator {
+
+    @Autowired
+    private EvaluationService evaluationService;
+
+    public EvaluatedPersonData evaluate(PersonDataDefinition definition, EvaluationContext context) throws EvaluationException {
+        EvaluatedPersonData c = new EvaluatedPersonData(definition, context);
+
+        String qry = "select\n" +
```
Comment (ojwanganto) [inline api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/ActivePatientsPopulationTypeDataEvaluator.java:37]: Can we also followup whether the KP forms should be used to determine this

Comment (patryllus) [reply api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/ActivePatientsPopulationTypeDataEvaluator.java:37]: Let me follow up with Kezia .... she had proposed the  use of hivGreencard and will also discuss with her situations where there are differing pop types in the two program followup forms

**Pull request closed without merging**